image_buffer/image_buffer.go

```diff
@@ -57,12 +78,12 @@ func (ib *ImageBuffer) MarkShouldSend(now time.Time) {
 		existingTimes[existing.Meta.CapturedAt.UnixNano()] = true
 	}
 
-	// Add images from the ring buffer that are within the window
-	windowDuration := ib.windowDuration()
 	for _, cached := range ib.ringBuffer {
-		timeDiff := triggerTime.Sub(cached.Meta.CapturedAt)
+		// timeDiff := triggerTime.Sub(cached.Meta.CapturedAt)
 		// Include images within windowSeconds before and after trigger
-		if timeDiff >= -windowDuration && timeDiff <= windowDuration {
+		// logger.Debugf("timeDiff", timeDiff.String())
+		logger.Debugf("cached.Meta.CapturedAt", cached.Meta.CapturedAt.String())
+		if !cached.Meta.CapturedAt.Before(ib.captureFrom) && !cached.Meta.CapturedAt.After(ib.captureTill) {
```

Comment:
I suspect you've phrased it like this because `.Before` and `.After` do not include the cases when the two times are equal, and we want to include those. Put in a comment explaining that.